Medipen refiller code improvement & QoL

JohnFulpWillard opened the pull request:
## About The Pull Request

- Turns the 'allowed' list into a static list and renames it to allowed_pens to give better explanation of what it is at first glance
- Adds context to the machine and its interactions.
- Replaces spaces in 'allowed' list with tabs
- Removes copy paste code between Initialize and RefreshParts
- Replaces single-letter vars
- Replaces to chat messages with balloon alerts
- Gives better feedback when machine fails to work properly
- Uses a do_after (so progress bar) when refilling a pen, instead of a magical timer. This also removed the need for the 'busy' var.
- Makes medipen refillers directly refill the medipen instead of deleting it and creating a brand new one.
- Removes parent calls from tool acts.

## Why It's Good For The Game

I didn't understand why they deleted medipens and respawned them, so I wanted to mainly fix that. Then when I looked at it, I noticed more and more problems, so I re-coded most of the machine to hopefully be better.

Because we now refill the medipen directly, it stays in the user's hand instead of teleporting to the tile of the refiller, which I also think is a positive for the same reason we changed vendors to automatically dispense into your hand.

The progress bar also gives more feedback to the user if the machine is/isn't working, which I find to be an improvement.

## Changelog

:cl:
qol: Medipen refillers now use balloon alerts and progress bars instead of to chat messages and invisible timers
code: Medipen refillers now refill your pen instead of delete it and create a new one. This also means that refilling them will leave the medipen in your hand instead of drop it at the refiller's tile.
/:cl:

JohnFulpWillard committed:
Medipen refiller code improvement & QoL

- Turns the 'allowed' list into a static list
- Removes copy paste code between Initialize and RefreshParts
- Replaces single-letter vars
- Replaces to chat messages with balloon alerts
- Gives better feedback when machine fails to work properly
- Uses a do_after (so progress bar) when refilling a pen, instead of a magical timer
- Makes medipen refillers directly refill the medipen instead of deleting it and creating a brand new one.
- Removes parent calls from tool acts.

JohnFulpWillard committed:
Merge branch 'master' into medipen-refiller

newfren commented:
does it still refill the sanguirite, formaldehyde, etc., that the epipens have in addition to epinephrine?

JohnFulpWillard commented:
> does it still refill the sanguirite, formaldehyde, etc., that the epipens have in addition to epinephrine?

yes

JohnFulpWillard committed:
Adds context to medipen refiller

Adds context to the medipen refiller machine, and also renames 'allowed' to 'allowed_pens' var to be more clear on what it is supposed to be.
Also adds a check for reagents when trying to transfer to the machine, so it doesn't state it's 'full' when it isn't.

JohnFulpWillard committed:
Use icon state instead of hardcoding it

JohnFulpWillard committed:
Doesn't use early return
